internal/provider/provider_internal_test.go

```diff
@@ -284,9 +284,6 @@ func Test_computeEnvFromOptions(t *testing.T) {
 				"FOO":                  "bar",    // should be included
 			},
 			expectEnv: map[string]string{
-				"CODER_AGENT_SUBSYSTEM":               "one,two",
-				"CODER_AGENT_TOKEN":                   "string",
-				"CODER_AGENT_URL":                     "string",
```

Comment:
These can simply be handled via `extra_env`. We don't need to test for them here.